### PR TITLE
222 appliccant email input with type text instead of email

### LINKED ISSUE #222: Appliccant Email input with type text instead of email
**Describe the Bug**
On the first screen of the A Class Seedlot registration, the Appliccant Email input field type is `text` when it should be `email`.

**Expected Behaviour**
The input type should be `email`.

**Actual Behaviour**
The input type is `text`, making it harder typing an email on small devices.

** Steps To Reproduce**
Steps to reproduce the behaviour:
1. Log in into SPAR
2. On the left menu, click **Seedlots**
3. Click **Register an A class seedlot**
4. Fill in **Applicant agency number**
5. On the next field **Applicant email address**, when viewing on smarphones, when focusing this field, you'll see the traditional keyboard, instead of the email keyboard.

**Screenshots**
No screenshots.

**Desktop (please complete the following information):**
 - OS: iOS and Android
 - Browser: all
 - Version: irrelevant

**Additional context**
No additional notes.

### PR DESCRIPTION
**Pull request by RMCampos**
# Description

Fix input type for Applicant Email Address. 

PS: No more email inputs on the project.

⚠️ Note that I had to update some test files in order of having them passing for me. Please, test if that works for everyone!

Fixes #222 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

This was tested locally and with my cellphone. Email Keyboard is now working.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-228-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-228-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-228-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/__test__/JSdomPatch.ts [hunk 1]
```diff
@@ -5,6 +5,6 @@ export default class FixJSDOMEnvironment extends JSDOMEnvironment {
     super(...args);
 
     // Reason: https://github.com/jsdom/jsdom/issues/3363
-    this.global.structuredClone = structuredClone;
+    this.global.structuredClone = (val: any) => JSON.parse(JSON.stringify(val.toString()));
```
Comment (craigyu) [inline frontend/src/__test__/JSdomPatch.ts:8]: Does `structuredClone` not work for you?

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx [hunk 2]
```diff
@@ -212,8 +212,11 @@ const ParentTreeStep = (
   };
 
   const renderColOptions = () => {
-    const toggleableCols = headerConfig
-      .filter((header) => header.isAnOption && header.availableInTabs.includes(currentTab));
+    let toggleableCols: Array<HeaderObj> = [];
+    if (Array.isArray(headerConfig)) {
```
Comment (craigyu) [inline frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx:216]: why is the type check needed? headerConfig will always be an array

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx [hunk 3]
```diff
@@ -377,7 +380,7 @@ const ParentTreeStep = (
                             <TableHead>
                               <TableRow>
                                 {
-                                  headerConfig.map((header) => (
+                                  (Array.isArray(headerConfig) && headerConfig.map((header) => (
```
Comment (craigyu) [inline frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx:383]: same as the comment from line 216

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx [hunk 4]
```diff
@@ -284,7 +287,7 @@ const ParentTreeStep = (
                       : null
                   }
                   {
-                    (state.notifCtrl[currentTab].showError && orchardsData.length === 0)
+                    (state.notifCtrl[currentTab]?.showError && orchardsData.length === 0)
```
Comment (craigyu) [inline frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx:290]: I can't think of a case where the `notifCtrl[currentTab]` is null

File: frontend/src/__test__/JSdomPatch.ts (hunk 1 above)
Comment (RMCampos) [reply frontend/src/__test__/JSdomPatch.ts:8]: Nope. Same error as before. `structuredClone` is not defined on tests.

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx (hunk 2 above)
Comment (RMCampos) [reply frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx:216]: On tests I have `TypeError: headerConfig.filter is not a function`.

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx (hunk 3 above)
Comment (RMCampos) [reply frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx:383]: Yep, same here as well. `TypeError: headerConfig.map is not a function`.

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx (hunk 4 above)
Comment (RMCampos) [reply frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/index.tsx:290]: On tests.

**Pull request merged**